### DISCUSSION TITLE
Convert SQLite to MySQL

### DISCUSSION THREAD

**Discussion by thunderclap82** [photoprism/photoprism#3505] [Q&A]
I have PhotoPrism running in docker on Unraid and have it configured to use SQLite. I want to switch to MariaDB and have tried the information at https://docs.photoprism.app/getting-started/advanced/backups/#sqlite-backups. The SQL file is created, however when I try to import it I'm getting the error

`ERROR 1064 (42000) at line 1: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'PRAGMA foreign_keys=OFF' at line 1`

Any ideas on how to fix this so I can import the SQL file? Thank you.

**Comment by lastzero** [comment]
AFAIK, `PRAGMA foreign_keys=off;` is SQLite syntax as MariaDB generally uses `SET FOREIGN_KEY_CHECKS=0`. So maybe you need to manually change or remove that line?

**Reply by thunderclap82** [reply]
That's helpful but then the syntax for line 2 is wrong, and then when I fix that the syntax for line 3 is wrong. I don't think the command on that page creates a proper SQL file, at least not for my version of MariaDB (10.11.4). Is it maybe for an older version?

**Reply by bergernetch** [reply]
> That's helpful but then the syntax for line 2 is wrong, and then when I fix that the syntax for line 3 is wrong. I don't think the command on that page creates a proper SQL file, at least not for my version of MariaDB (10.11.4). Is it maybe for an older version?

It is a proper SQL file - but just for the currently running DB engine. But SQL files are not freely exchangeable between databases, because these meta informations that are DB specific. The different standards are called SQL dialects.

There are specific tools and manuals on how to do this conversion. Some suggestions (not personally tested):

- [guide](https://www.geeksforgeeks.org/how-to-migrating-data-from-sqlite-to-mariadb/)
- [script ConvertFromSQLiteToMariadb](https://github.com/cpsource/ConvertFromSQLiteToMariadb)
- [online converter](https://www.rebasedata.com/convert-sqlite-to-mariadb-online)

**Reply by bergernetch** [reply]
Oh and maybe consult the official guide as well [Photoprism - Migrating from SQLite to MariaDB](https://docs.photoprism.app/getting-started/advanced/migrations/sqlite-to-mariadb/)
